jupyterhub_singleuser_profiles/profiles.py

```diff
@@ -226,7 +235,21 @@ def apply_gpu_config(self, gpu_mode, gpu_count, pod):
         if gpu_mode == self.GPU_MODE_PRIVILEGED:
           pod.spec.security_context.privileged = True
 
-    return pod
+      if gpus:
+        for gpu in gpus:
+          tolerations.append(gpu.get('node_tolerations')[0])
+    else:
+      if profile.get('node_tolerations'):
+         tolerations = profile.get('node_tolerations')
+
+      if profile.get('node_affinity'):
+          if not pod.spec.affinity:
+              pod.spec.affinity = {}
+          pod.spec.affinity['nodeAffinity'] = profile.get('node_affinity')
+
+    pod.spec.tolerations = tolerations
```

Comment:
Would it make sense to check the `pod.spec.tolerations` and if it is a `list`, then `append` instead of just set it?